assets/src/dashboard/components/tooltip/tooltip.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Internal dependencies
+ */
+import {
+  TOOLTIP_RTL_PLACEMENT,
+  Tooltip as BaseTooltip,
+  TooltipPropTypes,
+  TOOLTIP_PLACEMENT,
+} from '../../../design-system';
+import { useConfig } from '../../app/config';
+
+export default function Tooltip({
+  placement = TOOLTIP_PLACEMENT.BOTTOM,
+  ...props
+}) {
+  const { isRTL } = useConfig();
+  const derivedPlacement = isRTL ? TOOLTIP_RTL_PLACEMENT[placement] : placement;
```

Comment:
The base `Tooltip` component doesn't care whether the app is in LTR or RTL. The tooltip just needs to be told where to render the content.

This wrapper uses the config to decide where to place the tooltip content.

💭 Note: If we weren't using portals this may not be necessary. However, there's probably a use case that I don't know that requires us to use portals.

Reply:
I like simple tooltips as well. But portals are cool because you don't have to worry about your wrapped components hierarchy in the DOM. For example, let's say you wrapped your tooltip around your tool, but your tool is inside a container which hides overflowing elements, now the tooltip will be hidden by the container.

Idk if this is the case for tooltips in editor either, but that is one reason portals are used for implementing them.

Reply:
yea this was the case me and whale were talking about yesterday. Just a lot of complexity & runtime work added to mimic normal DOM behavior since we ripped it out of the DOM flow.

> but your tool is inside a container which hides overflowing elements

Yea we mentioned that this was the rationale for the decisions, but were considering how often does this scenario actually come up, and are the instances when it does come up so important that it's worth the added complexity? 

Anyway, it's definitely something to consider, remember & reassess if the editor designs ever start limiting the nested scrollable containers used in it

Reply:
Another approach I've considered here is just actually laying an absolutely positioned tooltip with opacity of 0 within the DOM flow, and just measuring that one node and applying it's measurements to an exact copy of it rendered in a portal (without the 0 opacity).

Would cut our DOM reads in half and get rid of a lot of the math logic in here. Ultimately tho, I don't think that would get rid of a lot of complexity in dynamically positioning the tail, and probably isn't worth the effort of a reimplementation cus the existing implementation is working fine

Reply:
>Another approach I've considered here is just actually laying an absolutely positioned tooltip with opacity of 0 within the DOM flow, and just measuring that one node and applying it's measurements to an exact copy of it rendered in a portal (without the 0 opacity).

I like this.

>Ultimately tho, I don't think that would get rid of a lot of complexity in dynamically positioning the tail, and probably isn't worth the effort of a reimplementation cus the existing implementation is working fine

But I agree with this as well.

I think the fewer DOM reads are what we want more than anything but the tooltips don't cause any visual glitches right now as you say. It's nice to keep in the toolbox for the future if we ever see something strange or really think the complexity trade off isn't worth making here.

Reply:
It could be interesting to explore this down the road with a little bit of free time as a potential perf win. I'll make a ticket.

Reply:
https://github.com/google/web-stories-wp/issues/6921